chat-android/src/main/java/com/ably/chat/Message.kt

```diff
@@ -68,10 +71,98 @@ data class Message(
      * Do not use the headers for authoritative information. There is no server-side
      * validation. When reading the headers treat them like user input.
      */
-    val headers: MessageHeaders,
+    val headers: MessageHeaders?,
 
     /**
      * The latest action of the message. This can be used to determine if the message was created, updated, or deleted.
+     * Spec: CHA-M10
      */
     val action: MessageAction,
+
+    /**
+     * A unique identifier for the latest version of this message.
+     * Spec: CHA-M10a
+     */
+    val version: String,
+
+    /**
+     * The timestamp at which this version was updated, deleted, or created.
+     */
+    val timestamp: Long,
+
+    /**
+     * The details of the operation that modified the message. This is only set for update and delete actions. It contains
+     * information about the operation: the clientId of the user who performed the operation, a description, and metadata.
+     */
+    val operation: Message.Operation? = null,
 )
+
+fun com.ably.chat.Message.copy(text: String, metadata: MessageMetadata? = null, headers: MessageHeaders? = null): com.ably.chat.Message =
```

Comment:
we don't need this right now, it won't be used anyway, it will be shadowed by the data class copy method

Reply:
Yeah, seems like it. Although, this raises questions about what type of fields user can exactly modify as a part of update : (
There's need to look at official doc for the same

Reply:
Okay, if you import `copy` method, original copy method doesn't overshadow it

Reply:
https://github.com/ably/ably-chat-kotlin/pull/100/commits/99a0fa27c48be1e7a56e1ea846ec6927a4d8a6cf

Reply:
I have removed overriden `copy` method for now. Though, innate `copy` method allows overriding `serial` which should be avoided at all costs. Not sure if we should encourage use of `copy` method, instead we might like to provide `clone` method which restricts number of params that needs to be updated.